de.dentrassi.pm.aspect.mvnosgi/src/de/dentrassi/pm/aspect/mvnosgi/internal/VirtualizerImpl.java

```diff
@@ -43,15 +42,41 @@ public void virtualize ( final Context context )
 
     private void processVirtualize ( final Context context ) throws Exception
     {
-        final ArtifactInformation art = context.getArtifactInformation ();
+        // The group ID will be set using the provided channel meta data or the default one.
+        String groupId = null;
+
+        // Extract some informations (e.g. group ID) from the provided channel meta data.
+        final Map<MetaKey, String> channelMetaData = context.getProvidedChannelMetaData ();
+        for ( final Entry<MetaKey, String> entry : channelMetaData.entrySet () )
```

Comment:
So this one actually belongs here:

True. Since it is backed by a TreeMap. However I think the whole bunch of IO operations far out weight the, rather quick, memory access ;-)

Do as you like ;-) more important is that things get done :-)